sdk-libs/sdk/src/compressible/decompress_idempotent.rs

```diff
@@ -0,0 +1,172 @@
+use anchor_lang::Discriminator;
+#[cfg(feature = "anchor")]
+use anchor_lang::{AnchorDeserialize as BorshDeserialize, AnchorSerialize as BorshSerialize};
+#[cfg(not(feature = "anchor"))]
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_compressed_account::{
+    address::derive_address, instruction_data::with_account_info::CompressedAccountInfo,
+};
+use light_hasher::DataHasher;
+use solana_account_info::AccountInfo;
+use solana_cpi::invoke_signed;
+use solana_msg::msg;
+use solana_pubkey::Pubkey;
+use solana_rent::Rent;
+use solana_system_interface::instruction as system_instruction;
+use solana_sysvar::Sysvar;
+
+use crate::{
+    account::LightAccount, compressible::compression_info::HasCompressionInfo, cpi::CpiAccounts,
+    error::LightSdkError, LightDiscriminator,
+};
+
+// #[cfg(not(feature = "anchor"))]
+// pub trait AnchorDiscriminatorShim {}
+// #[cfg(not(feature = "anchor"))]
+// impl<T> AnchorDiscriminatorShim for T {}
+
+/// Helper function to decompress multiple compressed accounts into PDAs
+/// idempotently with seeds. Does not invoke the zk compression CPI. This
+/// function processes accounts of a single type and returns
+/// CompressedAccountInfo for CPI batching. It's idempotent, meaning it can be
+/// called multiple times with the same compressed accounts and it will only
+/// decompress them once. If a PDA already exists and is initialized, it skips
+/// that account.
+///
+/// # Arguments
+/// * `pda_accounts` - The PDA accounts to decompress into
+/// * `compressed_accounts` - The compressed accounts to decompress
+/// * `signer_seeds` - Signer seeds for each PDA including bump (standard Solana
+///   format)
+/// * `cpi_accounts` - Accounts needed for CPI
+/// * `owner_program` - The program that will own the PDAs
+/// * `rent_payer` - The account to pay for PDA rent
+/// * `address_space` - The address space for the compressed accounts
+///
+/// # Returns
+/// * `Ok(Vec<CompressedAccountInfo>)` - CompressedAccountInfo for CPI batching
+/// * `Err(LightSdkError)` if there was an error
+pub fn prepare_accounts_for_decompress_idempotent<'info, T>(
+    pda_accounts: &[&AccountInfo<'info>],
```

Comment:
```suggestion
    solana_accounts: &[&AccountInfo<'info>],
```
would be more explicit

Reply:
renamed